client/components/Games/Level0/Game03.js

```diff
@@ -1,8 +1,111 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { Computer, GameButton, GameContent, GameText, Main } from '../../style';
+import styled from 'styled-components';
 
 const Game03 = () => {
-  return (<></>)
+  const [availButtons, setAvailButtons] = useState([
+    `catsTarget`,
+    `return`,
+    `cats++`,
+  ]);
+  const [usedButtons, setUsedButtons] = useState([`_____`, `_____`, `_____`]);
+  const winState = [`catsTarget`, `cats++`, `return`];
+  let won = false;
 
-}
+  if (availButtons.length === 0) {
+    won = (usedButtons.every((val, index) => val === winState[index]))
+  }
+
+  if (won){
+    console.log('you win!!');
+  }
+
+  const availClickHandler = (e) => {
+    const text = e.target.innerHTML;
+    //remove clicked button from available buttons
+    const newAvailButtons = availButtons.filter((a) => a !== text);
+    //add clicked button to used buttons in first "empty" location
+    const newUsedButtons = [...usedButtons];
+    for (let i = 0; i < newUsedButtons.length; i++) {
+      if (newUsedButtons[i] === `_____`) {
+        newUsedButtons[i] = text;
+        break;
+      }
+    }
+    setAvailButtons(newAvailButtons);
+    setUsedButtons(newUsedButtons);
+  };
+
+  const usedClickHandler = (e) => {
+    const text = e.target.innerHTML;
+    const index = e.target.id;
+    const newUsedButtons = [...usedButtons];
+    const newAvailButtons = [...availButtons];
+
+    if (text !== `_____`) {
+      newAvailButtons.push(text);
+      newUsedButtons[index] = `_____`;
+      setAvailButtons(newAvailButtons);
+      setUsedButtons(newUsedButtons);
+    }
+  };
+
+  const BiggerGameContent = styled(GameContent)`
+    height: 40vh;
```

Comment:
This is where the errors on my browser are coming from. If we move this outside of the component (up by line 4) the errors go away.

Reply:
oh good call on moving those out of the component function!